app/src/main/java/seemoo/fitbit/activities/WorkActivity.java

```diff
@@ -366,6 +380,18 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
         });
     }
 
+
+    /**
+     * {@inheritDoc}
+     *
+     */
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        requestPermissionsLocation();

```

Comment:
could we just put requestPermissionsLocation in onRestart of RequestPermissionsActivity ?